src/Microsoft.SqlTools.ServiceLayer/LanguageServices/LanguageService.cs

```diff
@@ -691,6 +710,76 @@ internal Hover GetHoverItem(TextDocumentPosition textDocumentPosition, ScriptFil
         }
 
         /// <summary>
+        /// Get function signature help for the current position
+        /// </summary>
+        internal SignatureHelp GetSignatureHelp(TextDocumentPosition textDocumentPosition, ScriptFile scriptFile)
+        {
+            int startLine = textDocumentPosition.Position.Line;
+            int startColumn = TextUtilities.PositionOfPrevDelimeter(
+                                scriptFile.Contents,    
+                                textDocumentPosition.Position.Line,
+                                textDocumentPosition.Position.Character);
+            int endColumn = textDocumentPosition.Position.Character;
+
+            ScriptParseInfo scriptParseInfo = GetScriptParseInfo(textDocumentPosition.TextDocument.Uri);
+
+            ConnectionInfo connInfo;
+            LanguageService.ConnectionServiceInstance.TryFindConnection(
+                scriptFile.ClientFilePath, 
+                out connInfo);
+
+            // reparse and bind the SQL statement if needed
+            if (RequiresReparse(scriptParseInfo, scriptFile))
+            {       
+                ParseAndBind(scriptFile, connInfo);
+            }
+
+            if (scriptParseInfo != null && scriptParseInfo.ParseResult != null)
+            {
+                if (Monitor.TryEnter(scriptParseInfo.BuildingMetadataLock))
+                {
+                    try
+                    {
+                        QueueItem queueItem = this.BindingQueue.QueueBindingOperation(
+                            key: scriptParseInfo.ConnectionKey,
+                            bindingTimeout: LanguageService.HoverTimeout,
```

Comment:
I would use BindingTimeout here instead of HoverTimeout...though they both are 500ms at this point, they *could* diverge in the future.